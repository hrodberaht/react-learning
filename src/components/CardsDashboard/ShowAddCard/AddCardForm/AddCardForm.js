import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addCard } from '../../../../actions';

export class AddCardForm extends Component {
  state = {
    name: '',
    email: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  resetState = () => {
    this.setState({
      name: '',
      email: '',
    });
  }

  handleAdd = (e) => {
    e.preventDefault();
    const { name, email } = this.state;
    const { handleAddCard } = this.props;
    handleAddCard(name, email);
    this.resetState();
  };

  render() {
    const { name, email } = this.state;
    return (

      <div className="AddCard">
        <form onSubmit={this.handleAdd}>
          <input
            id="name"
            type="text"
            value={name}
            onChange={this.handleChange}
          />
          Email:
          <input
            id="email"
            type="email"
            value={email}
            onChange={this.handleChange}
          />
          <button
            id="submit"
            type="submit"
            disabled={name.length < 3}
          >
          Add
          </button>
        </form>
      </div>
    );
  }
}

export default connect(
  null,
  { handleAddCard: addCard },
)(AddCardForm);

AddCardForm.propTypes = {
  handleAddCard: PropTypes.func.isRequired,
};
