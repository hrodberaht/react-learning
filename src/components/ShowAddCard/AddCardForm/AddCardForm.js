import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { addCard } from '../../../actions';

class AddCardForm extends Component {
  state = {
    name: '',
    email: '',
  };

  handleChange = (e) => {
    this.setState({ [e.target.id]: e.target.value });
  };

  handleAdd = (e) => {
    e.preventDefault();
    const { name, email } = this.state;
    const { handleAddCard } = this.props;
    handleAddCard(name, email);
    this.setState({
      name: '',
      email: '',
    });
  };

  render() {
    const { name, email } = this.state;
    return (

      <div className="AddCard">
        <form onSubmit={this.handleAdd}>
          <input
            id="name"
            type="text"
            value={name}
            onChange={this.handleChange}
          />
          Email:
          <input
            id="email"
            type="email"
            value={email}
            onChange={this.handleChange}
          />
          <button
            id="submit"
            type="submit"
            disabled={name.length < 3}
          >
          Add
          </button>
        </form>
      </div>
    );
  }
}

AddCardForm.propTypes = {
  handleAddCard: PropTypes.func.isRequired,
};

export { AddCardForm };
export const ConnectedAddCardForm = connect(
  null,
  { handleAddCard: addCard },
)(AddCardForm);
