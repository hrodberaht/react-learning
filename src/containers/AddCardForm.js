import { connect } from 'react-redux';
import { addCard } from '../actions';
import AddCardForm from '../components/ShowAddCard/AddCardForm/AddCardForm';


const mapDispatchToProps = dispatch => ({
  handleAddCard: card => dispatch(addCard(card.name, card.email)),
});


export default connect(null, mapDispatchToProps)(AddCardForm);
