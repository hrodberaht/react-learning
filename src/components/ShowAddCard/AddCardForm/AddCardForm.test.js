import { shallow } from 'enzyme';
import React from 'react';
import { AddCardForm } from './AddCardForm';

describe('<ShowAddCard />', () => {
  let wrapper;
  const props = {
    isAddCardVisible: false,
    handleAddCard: jest.fn(),
  };
  beforeEach(() => {
    wrapper = shallow(<AddCardForm {...props} />);
  });

  it('should render without crash', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should run function after click', () => {
    wrapper.find('button').simulate('click');
    expect(props.handleAddCard.mock.calls.length).toBe(1);
  });

  it('should state be empty strings', () => {
    expect(wrapper.state().name).toBe('');
    expect(wrapper.state().email).toBe('');
  });
  
});
