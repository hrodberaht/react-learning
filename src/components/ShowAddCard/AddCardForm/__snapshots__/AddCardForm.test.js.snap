// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowAddCard /> should render without crash 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddCardForm
    handleAddCard={[Function]}
    isAddCardVisible={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          htmlFor="name"
>
          Name:
          <input
                    id="name"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
</label>,
        <label
          htmlFor="email"
>
          Email:
          <input
                    id="email"
                    onChange={[Function]}
                    type="email"
                    value=""
          />
</label>,
        <button
          disabled={true}
          onClick={[Function]}
          type="button"
>
          Add
</button>,
      ],
      "className": "AddCard",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Name:",
            <input
              id="name"
              onChange={[Function]}
              type="text"
              value=""
/>,
          ],
          "htmlFor": "name",
        },
        "ref": null,
        "rendered": Array [
          "Name:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "name",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Email:",
            <input
              id="email"
              onChange={[Function]}
              type="email"
              value=""
/>,
          ],
          "htmlFor": "email",
        },
        "ref": null,
        "rendered": Array [
          "Email:",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Add",
          "disabled": true,
          "onClick": [Function],
          "type": "button",
        },
        "ref": null,
        "rendered": "Add",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            htmlFor="name"
>
            Name:
            <input
                        id="name"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</label>,
          <label
            htmlFor="email"
>
            Email:
            <input
                        id="email"
                        onChange={[Function]}
                        type="email"
                        value=""
            />
</label>,
          <button
            disabled={true}
            onClick={[Function]}
            type="button"
>
            Add
</button>,
        ],
        "className": "AddCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Name:",
              <input
                id="name"
                onChange={[Function]}
                type="text"
                value=""
/>,
            ],
            "htmlFor": "name",
          },
          "ref": null,
          "rendered": Array [
            "Name:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "name",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Email:",
              <input
                id="email"
                onChange={[Function]}
                type="email"
                value=""
/>,
            ],
            "htmlFor": "email",
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Add",
            "disabled": true,
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "Add",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
